src/providers/fault-store/fault-store.ts

```diff
@@ -79,61 +79,77 @@ export class FaultStoreProvider {
-   * screen.
-   *
-   * {@link TestSummaryComponent}
-   */
-  calculateFaultTotals(): void {
-    let drivingFaults = 0;
-    let dangerousFaults = 0;
-    let seriousFaults = 0;
-    this.seriousFaults = [];
-    this.dangerousFaults = [];
-    this.drivingFaults = [];
+  calculateFaultTotals(): ITestResults {
+
+    let drivingFaultsNum = 0;
+    let dangerousFaultsNum = 0;
+    let seriousFaultsNum = 0;
+    const seriousFaults = [];
+    const dangerousFaults = [];
+    const drivingFaults = [];
     const sectionRegEx = new RegExp(/^[a-z]*/);
     const data = this.getTotals();
     Object.keys(data.faults).forEach((fault) => {
       if (fault === 'lastFault') {
         return false;
       }
       if (data.faults[fault].fault) {
-        drivingFaults += data.faults[fault].fault;
+        drivingFaultsNum += data.faults[fault].fault;
         const section = upperFirst(fault.match(sectionRegEx)[0]);
-        this.drivingFaults.push({
+        drivingFaults.push({
           name: section + ' - ' + data.faults[fault].faultText,
           total: data.faults[fault].fault
         });
       }
       if (data.faults[fault].dangerous) {
-        dangerousFaults += data.faults[fault].dangerous;
+        dangerousFaultsNum += data.faults[fault].dangerous;
         const section = upperFirst(fault.match(sectionRegEx)[0]);
-        this.dangerousFaults.push({
+        dangerousFaults.push({
           name: section + ' - ' + data.faults[fault].faultText,
           total: data.faults[fault].fault
         });
       }
       if (data.faults[fault].serious) {
-        seriousFaults += data.faults[fault].serious;
+        seriousFaultsNum += data.faults[fault].serious;
         const section = upperFirst(fault.match(sectionRegEx)[0]);
-        this.seriousFaults.push({
+        seriousFaults.push({
           name: section + ' - ' + data.faults[fault].faultText,
           total: data.faults[fault].fault
         });
       }
     });
 
-    this.drivingFaultsNumber = drivingFaults;
-    this.seriousFaultsNumber = seriousFaults;
-    this.dangerousFaultsNumber = dangerousFaults;
+    this.drivingFaultsNumber = drivingFaultsNum;
+    this.seriousFaultsNumber = seriousFaultsNum;
+    this.dangerousFaultsNumber = dangerousFaultsNum;
+
+    const dangerousFaultSummary = {
+      title: FaultTitle.Dangerous,
+      total: dangerousFaultsNum,
+      faults: dangerousFaults
+    }
+    const seriousFaultSummary = {
+      title: FaultTitle.Serious,
+      total: seriousFaultsNum,
+      faults: seriousFaults
+    }
+    const drivingFaultSummary = {
+      title: FaultTitle.DriverFaults,
+      total: drivingFaultsNum,
+      faults: drivingFaults
+    }
+
+    return { dangerousFaultSummary, seriousFaultSummary, drivingFaultSummary }
 
   }
 
+  getFaultTotals(): Observable<ITestResults> {
+    return new Observable((observer: Observer<any>) => {
+      observer.next(this.calculateFaultTotals());
+      observer.complete();
+    });
```

Comment:
This works in the case we have here, but in the future if we rely on getFaultTotal to act as a dynamic observable I don't think this is going to work - Adding a new fault isn't going to re-trigger the calculation so your faultTotals are going to be wrong.
It is okay for now but if we rely on this for something like the running totals on the DL25 it won't work like we want it to.